site/content/troubleshooting-guide/index.md

```diff
@@ -67,4 +67,34 @@ AWS.Deploy.Tools, internally uses a variety of different services to host your .
+```
+The AWS CDK Bootstrap, which is the process of provisioning initial resources for the deployment environment, has failed. Please review the output above for additional details [and check out our troubleshooting guide for the most common failure reasons]. You can learn more about CDK bootstrapping at https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html.
+```
+The .NET Deployment Tool uses [AWS CDK](https://docs.aws.amazon.com/cdk/v2/guide/home.html) to create the AWS infrastructure needed to deploy your application. The AWS CDK is a framework for defining cloud infrastructure in code and provisioning it through AWS CloudFormation. 
+
+Deploying AWS CDK apps into an AWS environment (a combination of an AWS account and region) requires that you provision resources the AWS CDK needs to perform the deployment. These resources include an Amazon S3 bucket for storing files and IAM roles that grant permissions needed to perform deployments. The process of provisioning these initial resources is called [bootstrapping](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html).
+
+The required resources are defined in a AWS CloudFormation stack, called the bootstrap stack, which is usually named `CDKToolkit`. Like any AWS CloudFormation stack, it appears in the AWS CloudFormation console once it has been deployed.
+
+There could be several reasons why you are experiencing this issue. However, the most common ones are related to **Insufficient IAM Permissions** and an **Existing CDK Staging Bucket**. 
+
+**Possible Failure Reason**: When you try to deploy your application using the .NET Deployment Tool, your AWS account is checked for an existing `CDKToolkit` CloudFormation bucket. If it exists with the appropriate version, we continue with the deployment. If none exists, or you have an older version of it, then we try to create it or update the existing one. During that process, a set of IAM permissions are needed which typically include CloudFormation, S3 and ECR permissions. If you are missing some IAM permissions, that will cause the CDK Bootstrap to fail.
+
+**Resolution**: To fix the IAM Permissions issue, go through the deployment logs to spot which permissions the CDK Bootstrap failed on. This might look something like:
+```
+LookupRole API: iam:GetRole User: arn:aws:iam::123456789101:user/user is not authorized to perform: iam:GetRole on resource
+```
+
+This may require multiple iterations to determine all the needed permissions, but once you add the required IAM permissions you will be able to deploy. Checkout some recommended policies to setup your credentials [here](https://aws.github.io/aws-dotnet-deploy/docs/getting-started/setup-creds/).
+
+**Possible Failure Reason**: In some cases, the CDK Staging bucket gets created even though the CDK bootstrapping process failed. Therefore, the next time you try to run the CDK Bootstrap, it fails because the staging bucket already exists. The error could look something like:
```

Comment:
Replace with:
**Existing CDKToolkit S3 bucket**:  In rare cases, is it possible for the CDK Boostrap process not to clean the resources properly after the failed deployment. This causes the next deployment to fail as well, because the S3 bucket already exists. The error could look something like:
```
StagingBucket cdk-hnb659fds-assets-123456789101-us-west-2 already exists
```

Reply:
Done